src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -487,6 +487,8 @@ private class WaitDialogProgress : RestoreOperationProgressUI
                 JoinableTaskFactory jtf,
                 CancellationToken token)
             {
+                token.ThrowIfCancellationRequested();
+
                 await jtf.SwitchToMainThreadAsync();
```

Comment:
Throwing in case of cancellation requested I think would be more beneficial after switching to the main thread, or maybe checking before and after.

Imagine when StartAsync is called, the token is not cancelled, but it takes 1+ seconds to switch to the UI thread, during which time the token is cancelled. We wouldn't want to do "expensive" work on the UI thread in this case.

Reply:
We can also pass the token to `SwitchToMainThreadAsync(token)` .

https://github.com/microsoft/vs-threading/blob/main/doc/cookbook_vs.md#canceling-requests-for-the-main-thread

Then, we could have the following code pattern:

```c#
// ... some code 
token.ThrowIfCancellationRequested();

await jtf.SwitchToMainThreadAsync(token);

token.ThrowIfCancellationRequested();
// ... more code 
```

What do you think ?

Reply:
Added more calls.

Reply:
Passing the token to `SwitchToMainThreadAsync` will stop the `await` waiting for the result, but it won't prevent the continuation running on the main thead and using time. So, that `ThrowIfCancellationRequested` inside the main thread is still useful.